### PR TITLE
Prepare v0.51.0 release

### PR DESCRIPTION
**Pull request by johanfylling**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,64 @@
...
+- Dependency updates:
+  - build: bump actions/setup-go from 3 to 4 
+  - build: bump github.com/go-logr/logr from 1.2.3 to 1.2.4
+  - build: bump go.uber.org/automaxprocs from 1.5.1 to 1.5.2
+  - build: bump google.golang.org/grpc from 1.53.0 to 1.54.0
+  - build: bump oras.land/oras-go/v2 from 2.0.0 to 2.0.1
+  - build: bump oras.land/oras-go/v2 from 2.0.1 to 2.0.2
+  - tests: Remove testify dependency (authored by @charlieegan3)
+  
+### Don't promote?
```
Comment (johanfylling) [inline CHANGELOG.md:51]: My intention is to drop these from the changelog. LMK if some should be promoted.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,64 @@
 All notable changes to this project will be documented in this file. This
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 0.51.0
```
Comment (johanfylling) [inline CHANGELOG.md:6]: Previous releases are a bit more verbose. I'll write something up for the updates to monitoring, as I think we want to promote those.
Any other things to highlight? e.g. the `object.union_n` Wasm support?

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,64 @@
...
+
+### Monitoring
+
+- Surface unauthorized response count from OPA API authz handler ([#3378](https://github.com/open-policy-agent/opa/issues/3378)) authored by @ashutosh-narkar
+- plugins: Surface more decision log errors via status API ([#5637](https://github.com/open-policy-agent/opa/issues/5637)) authored by @ashutosh-narkar
+- plugins/rest: Include truncated HTTP response in logs ([#2961](https://github.com/open-policy-agent/opa/issues/2961)) authored by @ashutosh-narkar reported by @gshively11
+
+### Topdown and Rego
+
+- wasm: Add native support for `object.union_n` built-in function (authored by @Azanul)
```
Comment (srenatus) [inline CHANGELOG.md:18]: nit
```suggestion
- Wasm: Add native support for `object.union_n` built-in function (authored by @Azanul)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,64 @@
...
+
+### Fixes
+
+- ast/compile: Guard recursive module equality check. ([#5756](https://github.com/open-policy-agent/opa/issues/5756)) authored by @philipaconrad
```
Comment (srenatus) [inline CHANGELOG.md:22]: I'd drop the things shipped with previous patch releases, such as this one

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,7 +3,64 @@
...
+- ast: Relaxing strict-mode check for unused args in else-branching functions. ([#5758](https://github.com/open-policy-agent/opa/issues/5758)) authored by @johanfylling reported by @ethanjli
+- ast: Properly set the reported location of unused variables in strict-mode errors. ([#5662](https://github.com/open-policy-agent/opa/issues/5662)) authored by @boranx
+- fmt: report wrong arity for built-in functions. ([#5646](https://github.com/open-policy-agent/opa/issues/5646)) authored by @Trolloldem
+- topdown: http.send(): Ensuring intra-query caching consistency. ([#5736](https://github.com/open-policy-agent/opa/issues/5736)) authored by @johanfylling
+- Use Normalized policy paths as compiler module keys and store ids. (authored by @ashutosh-narkar)
```
Comment (srenatus) [inline CHANGELOG.md:27]: Wasn't this also patch release material? If not, we should mention what it fixed for Windows users.

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,7 +3,64 @@
...
+
+- Update -debug images to use Chainguard images ([5544](https://github.com/open-policy-agent/opa/issues/5544)) (authored by @charlieegan3)
+- Dependency updates:
+  - build: bump actions/setup-go from 3 to 4 
+  - build: bump github.com/go-logr/logr from 1.2.3 to 1.2.4
+  - build: bump go.uber.org/automaxprocs from 1.5.1 to 1.5.2
+  - build: bump google.golang.org/grpc from 1.53.0 to 1.54.0
+  - build: bump oras.land/oras-go/v2 from 2.0.0 to 2.0.1
+  - build: bump oras.land/oras-go/v2 from 2.0.1 to 2.0.2
+  - tests: Remove testify dependency (authored by @charlieegan3)
```
Comment (srenatus) [inline CHANGELOG.md:49]: I'd drop all of these dependency details. We could include something like "various third-party dependencies were updated."

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,7 +3,64 @@
...
+  - fixes a bug present in `0.50.1`
+- runtime: Remove deprecation warning for watch flag (authored by @tsandall)
+- Fixes to #5774
+  - [docs] Address anchor linking issue (#5795) (authored by @charlieegan3)
+  - [docs] Fix dashboard height issue (#5799) (authored by @charlieegan3)
+- Update retract directive comment (#5780) (authored by @charlieegan3)
+- download: copy bundle into buffer only if needed (#5767) (authored by @srenatus)
+- fix: preserve `jsonOptions` in NewAnnotationsRef (#5753) (authored by @zregvart)
+- mask: don't rely on JSON roundtrip to convert structs (#5770) (authored by @srenatus)
+- plugins/logs: don't convert EventV1 to ast.Value twice (#5761) (authored by @srenatus)
```
Comment (srenatus) [inline CHANGELOG.md:63]: We could mention "performance improvements to decision logging"

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,7 +3,64 @@
...
+- ast: Properly set the reported location of unused variables in strict-mode errors. ([#5662](https://github.com/open-policy-agent/opa/issues/5662)) authored by @boranx
+- fmt: report wrong arity for built-in functions. ([#5646](https://github.com/open-policy-agent/opa/issues/5646)) authored by @Trolloldem
+- topdown: http.send(): Ensuring intra-query caching consistency. ([#5736](https://github.com/open-policy-agent/opa/issues/5736)) authored by @johanfylling
+- Use Normalized policy paths as compiler module keys and store ids. (authored by @ashutosh-narkar)
+
+### Docs, Website, and Ecosystem
+
+- Fix typo in documentation (authored by @eternaltyro)
+- Update TLS authentication docs (authored by @charlieegan3)
+- Update _index.md (authored by @Ronnie-personal)
```
Comment (srenatus) [inline CHANGELOG.md:33]: Let's mention that the link for the windows sha sums was explicitly mentioned or something like that.

File: CHANGELOG.md (hunk 4 above)
Comment (johanfylling) [reply CHANGELOG.md:22]: Yep, missed that one, tnx :)

File: CHANGELOG.md (hunk 5 above)
Comment (johanfylling) [reply CHANGELOG.md:27]: You're right. I'll drop this one too 👍

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,7 +3,64 @@
...
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 0.51.0
+
+This release contains improvements to monitoring and an assortment of fixes and improvements.
+
+### Monitoring
+
+- Surface unauthorized response count from OPA API authz handler ([#3378](https://github.com/open-policy-agent/opa/issues/3378)) authored by @ashutosh-narkar
```
Comment (anderseknert) [inline CHANGELOG.md:12]: Surface where? :) Also maybe change "API" to "HTTP API" or "REST API"

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,7 +3,42 @@
...
+- plugins: Surface more decision log errors via status API ([#5637](https://github.com/open-policy-agent/opa/issues/5637)) authored by @ashutosh-narkar
+- plugins/rest: Include truncated HTTP response in logs ([#2961](https://github.com/open-policy-agent/opa/issues/2961)) authored by @ashutosh-narkar reported by @gshively11
+
+### Topdown and Rego
+
+- Wasm: Add native support for `object.union_n` built-in function (authored by @Azanul)
+
+### Fixes
+
+- ast: Relaxing strict-mode check for unused args in else-branching functions. ([#5758](https://github.com/open-policy-agent/opa/issues/5758)) authored by @johanfylling reported by @ethanjli
```
Comment (anderseknert) [inline CHANGELOG.md:22]: This was in a previous patch, right?

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,7 +3,42 @@
...
+- ast: Properly set the reported location of unused variables in strict-mode errors. ([#5662](https://github.com/open-policy-agent/opa/issues/5662)) authored by @boranx
+- fmt: report wrong arity for built-in functions. ([#5646](https://github.com/open-policy-agent/opa/issues/5646)) authored by @Trolloldem
+- topdown: http.send(): Ensuring intra-query caching consistency. ([#5736](https://github.com/open-policy-agent/opa/issues/5736)) authored by @johanfylling
+
+### Docs, Website, and Ecosystem
+
+- Fix typo in documentation (authored by @eternaltyro)
+- Update TLS authentication docs (authored by @charlieegan3)
+- Clarification in docs about checksums of Windows executables (authored by @Ronnie-personal)
+- docs: Small fix to context placement in integration (authored by @craigpastro)
```
Comment (anderseknert) [inline CHANGELOG.md:32]: It doesn't matter, but I'm curious as to why there are no links to the above changes.

File: CHANGELOG.md [hunk 12]
```diff
@@ -3,7 +3,42 @@
...
+- docs/website: Fix floating navbar anchor issue ([5774](https://github.com/open-policy-agent/opa/issues/5774)) authored by @charlieegan3 reported by @kristiansvalland
+- Ecosystem:
+  - Add Quali Torque (authored by @shirabendor-quali)
```
Comment (anderseknert) [inline CHANGELOG.md:35]: I think we can skip including ecosystem updates, but if we do, I think one line is enough here.

File: CHANGELOG.md [hunk 13]
```diff
@@ -3,7 +3,42 @@
...
+  
+### Miscellaneous
+
+- Update -debug images to use Chainguard images ([5544](https://github.com/open-policy-agent/opa/issues/5544)) (authored by @charlieegan3)
+- Various third-party dependencies were updated.
+- Performance improvements to decision logging
```
Comment (anderseknert) [inline CHANGELOG.md:41]: This seems like something I'd want to read more about, so a link would be good! Also, I'd move this to the Fixes section unless we'll have one for performance,

**Review by anderseknert — DISMISSED** [review]
LGTM. Left some comments but nothing blocking, so feel free to ignore.

File: CHANGELOG.md (hunk 11 above)
Comment (johanfylling) [reply CHANGELOG.md:32]: Some of the generated entries simply didn't pick up on the linked issue. I've attempted to manually correct for those; but some were simply just a "naked" PR. I'll go through these again to see if I've missed any that can be linked to an issue.

File: CHANGELOG.md (hunk 12 above)
Comment (johanfylling) [reply CHANGELOG.md:35]: We've listed them like this in the [past](https://github.com/open-policy-agent/opa/releases/tag/v0.50.0), even for single updates. But I'm fine with dropping this altogether, as I'm guessing this is of little consequence to most readers of the changelog.

File: CHANGELOG.md (hunk 13 above)
Comment (srenatus) [reply CHANGELOG.md:41]: It's not much, just removed some superfluous work in that code path... we don't have hard numbers either... so we might as well drop it 🤷

Comment (johanfylling) [reply CHANGELOG.md:41]: This is one of those PR:s with no attached issue; but perhaps @srenatus could come up with a short blurb describing the fix?

Comment (srenatus) [reply CHANGELOG.md:41]: "remove superfluous json encoding roundtrip and double work in AST conversion of to-be-logged events" ?

**Review by srenatus — APPROVED** [review]

**Pull request merged**